Functions/Quantification/xASL_quant_ASL.m

```diff
@@ -105,10 +105,17 @@
 PWI4D = double(PWI4D);
 M0_im = double(M0_im);
 
+% Create path to PWI3D based on the PWI4D path (Because PWI4D_Path can be a path to a merged NII)
+PWI3D_Path = PWI4D_Path;
+indexString = regexp(PWI4D_Path,'PWI4D');
+indexString = indexString(end); % Take the last occurence
+PWI3D_Path(indexString+3) = '3'; % Replace PWI4D by PWI3D
+
 if ~x.modules.asl.ApplyQuantification(3)
     fprintf('%s\n','We skip the scaling of a.u. to label intensity');
 	% In case we don't run quantification, we still need to average the PWI image
-	PWI = xASL_im_ASLSubtractionAveraging(x, [], PWI4D_Path);
+	xASL_io_ASLSubtractionAveraging(x, {2, PWI3D_Path}, 0, [], PWI4D_Path);
```

Comment:
Why don't you replace this with 

PWI = xASL_im_ASLSubtractionAveraging(x, [], xASL_io_Nifti2Im(PWI4D_Path));

I say this, because in the future I would like to simplify this:
1. load PWI4D
2. run BASIL or not
3. do the scalings
4. average the 4th dim (which would have no effect if it was already 1)
-> do you agree that we do this in #1572 ?

Reply:
Henk. You have just asked me to replace the IM by IO. Now you are asking me to switch back.

Moreover, we can't do this, because by loading PWI4D, we don't have information anymore about the PLDs as these are in the JSON. We can't use x.Q, because PWI4D here can be normal, despiked or merged.

Reply:
No only of course because at this point, the PWI4D and sidecars have been loaded already in the beginning of the function.

Reply:
Regarding xASL_io_ASLSubtraction - we have to use IO and not IM here, because we need to update PWI3D based on a new PWI4D. So this is not about loading JSON, but about saving new JSON+NII, so the IO has to be used as IM wouldn't save anything.

So we can't modify the code here. And we can do the rest in #1572.

Reply:
There is no new PWI4D. This is xASL_quant_ASL, which at the start of the function reads NIfTI and JSON, and not here.

There is a wrapper outside this function that does all the management. But this function should not do this, we are not saving intermediate steps here.

We can do this later. But again:

we have whatever wrappers that:
1. load PWI4D
2. load JSON
3. Manage basil/xASL quantification options bla bla
4. start xASL_quant_ASL -> which does the quantification
5. Save CBF

We do not save PWI3D intermediate, we do that in xASL_wrp_Resample.

Or are you saying that we have to save PWI3D again here, because we have potentially merged PWI4D or changed it or anything in between xASL_wrp_Resample and xASL_wrp_Quantify?

Then I agree, but then still it would be nice to have a clean management/wrapper and a single quantification function.

We can continue this conversation in #1572 indeed.

So you can resolve.

Reply:
Yes, we need new PWI3D. We had PWI4D, PWI3D and PWI calculated and saved previously. But then we do despiking and potentially exclude some volumes. So we need to redo PWI3D based on the PWI4D cause some volumes might be removed. So we need saving again. 

Just explaining - you can resolve.